algorithms/glued_trees/glued_trees.ipynb

```diff
@@ -0,0 +1,457 @@
+{
```

Comment:
<p>Reading more carefully this part, I'm not sure this is accurate:</p><ol><li>I think that what you are missing the 'momentum' coordinates in the state vector. Then the hamiltonian you want to simulate is ((0, B), (B_dag, 0)) (see eq. 12 in the paper). Then you also want a state preparation that initializes the momentum of the first coordinate to 1 (to have non-zero energy). See fig.4 in the paper for more intuition.</li><li>As far as I understand, the matrix A you simulate is an effective matrix that is a result of coordinate change, but does not represent the "real oracle". In the oracular setting, the state-space should be N x N, including all nodes. I think the the effective A is used in the paper for the proof. The effective space is small, but the number of used qubits should be still large (and not log-log of the number of nodes). The point is that you don't know beforehand the label of the exit node, just as the oracle builder.</li><li>You can choose, for demonstration purposes, an easy ordering of the nodes for the oracle, such as linear ordering</li></ol>
<br/>

 _Reply via <a href='https://app.reviewnb.com/Classiq/classiq-library/pull/151/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='2'/>